pkg/codegen/pcl/binder_schema.go

```diff
@@ -135,9 +138,68 @@ func canonicalizeToken(tok string, pkg schema.PackageReference) string {
 	return fmt.Sprintf("%s:%s:%s", pkg.Name(), pkg.TokenToModule(tok), member)
 }
 
+// getPkgVersion gets the package version from an unbounded resource node.
+func (b *binder) getPkgVersion(node *Resource, pkg string) string {
```

Comment:
Since we haven't called `bindNodes` yet, we aren't able to get the version from the resource node's `options` field unfortunately, so this is a workaround @AaronFriel helped direct me towards. One option could be to do a deep copy of the node so we don't mutate the original, but doing a partial bind here doesn't seem to affect the full binding afterwards.